Block/Menu.php

```diff
@@ -307,15 +307,16 @@ private function getMenuNodeBlock($node)
 
         $level = $node->getLevel();
         $isRoot = 0 == $level;
-
+        
         $nodeBlock->setId($node->getNodeId())
             ->setTitle($node->getTitle())
             ->setLevel($level)
             ->setIsRoot($isRoot)
             ->setIsParent((bool) $node->getIsParent())
             ->setIsViewAllLink(false)
             ->setContent($node->getContent())
-            ->setMenuClass($this->getMenu()->getCssClass());
+            ->setMenuClass($this->getMenu()->getCssClass())
+            ->setMenuCode($this->getData('menu'));
```

Comment:
To avoid multiple setters such us `setMenuClass`, `setMenuCode`, `setMenuXyz` consider passing the whole menu object `->setMenu($this->getMenu())`.

Reply:
I use here only two params from menu object I think we can leave instead add all menu If we need add more setters we can do like You said what You think ?

Reply:
okay